### ISSUE TITLE
Support for History API

### ISSUE THREAD

**Issue by ykamei** [kainsavage/HistoryBlock#38]
Thanks for the great add-on. I use this add-on conveniently every day.

How can I make it so that it doesn't keep the history of sites that use the HTML5 History API, like Gmail(maybe), etc.?
More and more sites are using history.pushState, so I would appreciate your support.
https://developer.mozilla.org/en-US/docs/Web/API/History_API/Working_with_the_History_API

**Comment by kainsavage** [comment]
Hmm, that's interesting. I will have to give that some thought, but the `closeTab` action should clear that out.

I'm sure that I can get a hook onto those APIs and clear them out in real time, but I will have to play with it for a bit.

**Comment by kainsavage** [comment]
So, after a bit of digging, I think that this is already supported, though you need some technical background to make it work.

tl;dr - use "full URL" matching, and add `mail.google.com/mail/u/0/”` - this will block gmail the way you describe.
